pyannote/audio/pipelines/speaker_diarization.py

```diff
@@ -530,6 +528,19 @@ def apply(
         # hard_clusters: (num_chunks, num_speakers)
         # centroids: (num_speakers, dimension)
 
+        # number of detected clusters is the number of different speakers
+        num_different_speakers = centroids.shape[0]
+        # quick sanity check
+        assert (
+            num_different_speakers >= min_speakers
+            and num_different_speakers <= max_speakers
+        )
```

Comment:
Does that `assert` ever fail?

If it does, we should handle it or this will raise an error.

Reply:
@hbredin Good catch. With the removal of the k-means fallback which used to be included in this PR (and which I use in my application anyway), this can definitely fail, and surely will in some circumstances due to the `min_cluster_size` parameter. The best we can do here I think is to issue a warning.